Core/StitchCoreSDK/Sources/StitchCoreSDK/Auth/Internal/CoreStitchAuth+OtherAuthActions.swift

```diff
@@ -0,0 +1,177 @@
+
+        // If the user is not logged in --> return
+        guard authInfo.isLoggedIn else { return }
+
+        // If it is an AnonymousUser remove the user altogether from the list
+        // Otherwise set the authInfo to have no access token or refresh token
+        var newAuthInfo = StoreAuthInfo.init(withAuthInfo: authInfo, withLogout: true).toAuthInfo
+        newAuthInfo = StoreAuthInfo.init(withAuthInfo: newAuthInfo, withNewTime: true).toAuthInfo
+
+        if authInfo.loggedInProviderType == .anonymous {
+            loggedInUsersAuthInfo.remove(at: index)
+        } else {
+            loggedInUsersAuthInfo[index] = newAuthInfo
+        }
+
+        try writeCurrentUsersAuthInfoToStorage(
+            currentUsersAuthInfo: loggedInUsersAuthInfo,
+            toStorage: storage)
+
+        // If this was the active user then set the activeUser to nil and clear its authState
+        if userId == activeUserAuthInfo?.userID {
+            // This will call onAuthEvent
+            clearActiveAuthInfo(storage: storage)
+        }
+
+        // Issue the logout request
+        _ = try? self.doLogout(authInfo)
+    }
+
+    /**
+     * Leaves the active user as logged in but makes the user with the given id the active user
+     */
+    public func switchToUserInternal(withId userId: String) throws -> TStitchUser {
+        objc_sync_enter(authOperationLock)
+        defer { objc_sync_exit(authOperationLock) }
+
+        // Get the user objects if it exists --> otherwise throw
+        guard let index = loggedInUsersAuthInfo.firstIndex(where: {$0.userID == userId}) else {
+            throw StitchError.serviceError(
+                withMessage: "User with id: \(userId) not found",
+                withServiceErrorCode: .userNotFound)
+        }
+        let authInfo = loggedInUsersAuthInfo[index]
+
+        if !authInfo.isLoggedIn {
+            throw StitchError.serviceError(
+                withMessage: "User with id: \(userId) not logged in",
+                withServiceErrorCode: .userNotLoggedIn)
+        }
+
+        // Trigger Auth Event
+        defer { onAuthEvent() }
+
+        // Update the lastAuthActivity of the old and new active user and persist
+        if let oldActiveAuthInfo = activeUserAuthInfo {
+            if let oldIndex = loggedInUsersAuthInfo.firstIndex(where: {$0.userID == oldActiveAuthInfo.userID}) {
+                let oldAuthInfo = StoreAuthInfo.init(withAuthInfo: oldActiveAuthInfo, withNewTime: true).toAuthInfo
+                loggedInUsersAuthInfo[oldIndex] = oldAuthInfo
+            }
+        }
+        let newAuthInfo = StoreAuthInfo.init(withAuthInfo: authInfo, withNewTime: true).toAuthInfo
+        loggedInUsersAuthInfo[index] = newAuthInfo
+        do {
+            try writeCurrentUsersAuthInfoToStorage(currentUsersAuthInfo: loggedInUsersAuthInfo, toStorage: storage)
+        } catch {
+            // Do nothing, would prefer not to fail here 
```

Comment:
[minor] you should throw a `couldNotPersistAuthInfo` client error here. Although rare, an error like this indicates that the system is in a really weird state and probably shouldn't be doing more stuff.